### DISCUSSION TITLE
About the documentation

### DISCUSSION THREAD

**Discussion by Caralire** [kudrykv/latex-yearly-planner#34] [General]
Hi @kudrykv 

First of all thank you for making and sharing this yearly planner.

I am about to get an ONYX BOOX note air 2 and was considering building such a document with LaTeX. I found about your project while looking for a base to maybe not start from scratch. But although it is named latex-yearly-planner, I realise it is much more than a LaTeX document : a mix of go, smarty, shell, and python, if I get it right.

As a math teacher, I am an educated amateur of Latex, and I know some basic Python. But I never heard of go, smarty or shell before yesterday. 

I installed go this morning, and am slowly going through initiation process. I would consider it a first success to manage to compile your code as is. This is where the documentation you're planning to write would come handy, cause right now I don't event know which is the main file. So I am writing this to let you know that at least me would read it, if you ever manage to write it.

My second achievement target is to manage to modify the beginning and end date, cause as I said, I'm a teacher, and I work on a school year beginning in September...

So, thanks again for everything, I will follow up the updates on your project.

**Comment by kudrykv** [comment]
Hi @Caralire!

I appreciate your interest in this project!

I understand your struggle. If I were you, it would take much effort to figure out how the project works, even having the knowledge of Go.

---

Here's a quick dive-in into what's going on:
1. Go (programming language) generates tex files.
2. Shell calls pdflatex to compile tex files into a final PDF.

Python script there is irrelevant and can be ignored. Smarty type is Github misjudging the file type (likely my mistake). The intention there was to have template files that I use to build tex files. Take a look at files in `tpl` folder: you'll see those are tex snippets with Go placeholders sprinkled over them.

The reason for having a programming language over tex is to have a way to programmatically generate all the pages (and there are plenty of them), make links between pages, etc. That is simply so much easier rather than writing everything in TeX.

If you want to try and build it locally, you need (roughly):
- the latest version of Go;
- the latest version of latex, pdflatex.

You need to create an `out` folder. Then, all you need to do is run this command in the shell:
```sh
PLANNER_YEAR=2022 \
PASSES=1 \
CFG="cfg/base.yaml,cfg/template_breadcrumb.yaml,cfg/sn_a5x.breadcrumb.default.yaml" \
NAME="sn_a5x.breadcrumb.default" \
./single.sh
```
(has to work with Linux and Mac; for Windows, one must read the internals of `single.sh` and replicate the behavior)

The command should just work without any interventions, and produce a PDF along to the `out` folder.

`single.sh` is a flow for building a single PDF. `release.sh` is a flow to build PDFs with all available permutations. It also uses `single.sh` when it builds PDFs.

**Reply by Caralire** [reply]
So quick!!

Thank you @kudrykv, I try to make my way through it now.

**Reply by kudrykv** [reply]
> My second achievement target is to manage to modify the beginning and end date, cause as I said, I'm a teacher, and I work on a school year beginning in September...

This one would be a blocker. I've thought about this before but didn't put an implementation. The concept of defining a custom "year window" was too much for the planner at that time. Please create an issue for that; I'll put that somewhere on the roadmap. I need to tell you, though, that it won't be fast unless you try to tackle it yourself.

The most straightforward thing here I can suggest here with some manual intervention:
1. Compile one year (e.g., for 2022).
2. Store those files separately.
3. Compile another year (e.g., 2023).
4. Manually edit and merge those files: take what you need from both, remove redundant.
5. Compile with pdflatex the PDF.

What I mean with p.4 is that you'll have a bunch of files (`annual.tex`, `quarterlies.tex`, etc). You'll need to manually go through them. For example, in `annual.tex` you will need to remove months from 2022 till September, get months till September from 2023, combine and put those back into `annual.tex`. Then, repeat the process for all other files.

While that would be relatively easy for `annual.tex` or `quarterlies.tex`, it would become harder and harder for the next files (`monthlies.tex`, `weeklies.tex`, etc), as they would be longer and longer. Those files are massive; it would be easy to make a mistake with the manual edit.

**Reply by Caralire** [reply]
Thanks again for everything.

I'll let you know how I manage to do that.

Have a nice day.

**Comment by bobby-abe** [comment]
Hi @kudrykv

I am having an issue running the build. I know just enough about linux to get most things installed, but typically am never building anything. So maybe it's just a path issue, not sure. But I've confirmed I have the latest versions of everything installed and run the various commands in the shell. Everything seems to work until I run the build line, upon where I get this. Any help would be appreciated. I want to build up future PDFs but wanted them all to have lines instead of dots. :)

bob@bob-linuxmint:~/Desktop/latex-yearly-planner-main$ ./single.sh
Incorrect Usage. flag needs an argument: -config

NAME:
   plannergen - A new cli application

USAGE:
   plannergen [global options] command [command options] [arguments...]

COMMANDS:
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --config value  
   --preview       (default: false)
   --help, -h      show help (default: false)
flag needs an argument: -config
exit status 1

**Reply by kudrykv** [reply]
It's a bunch of flags that are required
Try this:
```sh
PLANNER_YEAR="2023" \
PASSES="2" \
CFG="cfg/base.yaml,cfg/template_months_on_side.yaml,cfg/sn_a5x.mos.default.yaml,cfg/sn_a5x.mos.default.dailycal.yaml" \
NAME="sn_a5x.mos.default.dailycal.2023" \
./single.sh
```

**Comment by bobby-abe** [comment]
Two things.

1.) I now know that those lines are meant to all be cut/pasted as one line.
That wasn't clear to me in any instructions but now I know. :)

2) There were some required packages missing for some reason, but the
process prompted me and just downloaded whatever it needed. :)

So yeah, got it working!! :)
THANK YOU :)

**Reply by kudrykv** [reply]
Regarding p.1 — that's weird: backslashes at the end precisely to allow multiline input. So the copy-paste had to work; not sure why it didn't 🤔

I'm glad that in the end it worked, though!

**Comment by bobby-abe** [comment]
Should the output folder be cleaned up between generating each file? All
those aux and tex files?

**Reply by kudrykv** [reply]
It should be fine to keep them. LaTeX will handle that.

**Comment by bobby-abe** [comment]
The pages still have dots instead of lines. Is there a flag or parameter
that changes that?

**Reply by kudrykv** [reply]
Ah, sorry, I've missed that you wanted a lined one.

In the `cfg/base.yaml` file, change `dotted: true` to `dotted: false`

**Comment by bobby-abe** [comment]
Thanks so much! :)

**Comment by elementalcoder** [comment]
hey there, 
Thanks for the amazing project. Is there any way to render from a specific date to another date? for example I want to use this for the academic semester that starts from 20 Oct 2023 to 24 Jan 2024.
I would really appreciate it if there is such option to make a planner for this specific period.

**Reply by kudrykv** [reply]
Hi!
This is a work in progress. There will be a way to specify start and end months https://github.com/kudrykv/latex-yearly-planner/blob/62428a24757ad063f687be01d8f1b71d9a2a73ca/config/experimental.yaml#L55-L56

It's not really complete, though. I still need to implement links, etc. "Breadcrumbs" template, which makes more sense for such division, I didn't start yet.

Unfortunately, I have nothing to offer at this moment. Maybe, in a month or two.

**Comment by Boshimanu** [comment]
Hello guys, 
I saw from the example folder that I can add ToDo pages to my planner, but I cannot understand how to do that, I've spent like a 60-90 min to do this. In the end I've came to ask you how to do this... I've failed...😔
This is the command that I'm using:

PLANNER_YEAR=2025 \
PASSES=3 \
CFG="cfg/base.yaml,cfg/template_breadcrumb.yaml,cfg/sn_a6x.mos.default.yaml" \
NAME="4sn_a6x.mos.default" \
./single.sh

Any help would be appreciated.
🙏🙏🙏
